worlds/seaofthieves/Items/ItemAdder.py

```diff
@@ -0,0 +1,64 @@
+from ..Configurations import SotOptionsDerived
+from BaseClasses import MultiWorld, Region, Entrance, ItemClassification
+from ..Items.Items import Items, SOTItem
+from ..Items.ItemCollection import ItemCollection
+import math
+
+
+def create_items(world: MultiWorld, location_count: int, options: SotOptionsDerived, itemCollection: ItemCollection,
+                 player: int):
+    update_items_from_options(options, itemCollection)
+
+    # we know the pre fill locations cannot be used
+    location_count = location_count - itemCollection.pre_fill_count
+
+    location_count -= __add_main_items(world, itemCollection, player)
+    location_count -= __add_fill_and_traps(world, itemCollection, player, location_count, options)
+
+    if (location_count != 0):
+        raise Exception(
+            "In item creation, items added to the pool resulted in {} locations remaining.".format(location_count))
```

Comment:
```suggestion
    if (location_count != 0):
        raise Exception(
            f"In item creation, items added to the pool resulted in {location_count} locations remaining.")
```
Would recommend just using an f string here